### PR TITLE
Solve Issues

### LINKED ISSUE #190: [Feature Request] Add convenience function to access the text value of a CompositeCell
**Is your feature request related to a problem? Please describe.**
See https://github.com/nfdi4plants/ARCtrl/pull/189/files#r1323416055 , just tracking here so it is not forgotten.

### LINKED ISSUE #197: [Feature Request] Creating CompositeHeader on JS side needs API support

### LINKED ISSUE #198: [BUG] ArcTable.UpdateHeader forceConvert flag updates cells not smart enough
```
member this.UpdateHeader (index:int, newHeader: CompositeHeader, ?forceConvertCells: bool) =
        let forceConvertCells = Option.defaultValue false forceConvertCells
        ArcTableAux.SanityChecks.validateColumnIndex index this.ColumnCount false
        /// remove to be replaced header, this is only used to check if any OTHER header is of the same unique type as column.Header
        /// MUST USE "Seq.removeAt" to not remove in mutable object!
        let otherHeaders = this.Headers |> Seq.removeAt index
        ArcTableAux.SanityChecks.validateNoDuplicateUnique newHeader otherHeaders
        let c = { this.GetColumn(index) with Header = newHeader }
        // Test if column is still valid with new header, if so insert header at index
        if c.validate() then
            let setHeader = this.Headers.[index] <- newHeader
            ()
        // if we force convert cells, we want to convert the existing cells to a valid cell type for the new header
        elif forceConvertCells then
            let convertedCells =
                match newHeader with
                | isTerm when newHeader.IsTermColumn -> c.Cells |> Array.map (fun c -> c.ToTermCell())
                | _ -> c.Cells |> Array.map (fun c -> c.ToFreeTextCell())
            this.UpdateColumn(index, newHeader, convertedCells)
        else
            failwith "Tried setting header for column with invalid type of cells. Set `forceConvertCells` flag to automatically convert cells into valid CompositeCell type."
```

### PR DESCRIPTION
**Pull request by Freymaurer**
- closes #198 
- closes #190 
- closes #197

### CODE REVIEW

File: tests/TestingUtils/Library.fs [hunk 1]
```diff
@@ -45,41 +45,39 @@ module Result =
...
         | Error m -> m
 
-
-
 /// Fable compatible Expecto/Mocha unification
 module Expect = 
     open Utils
 
     let inline equal actual expected message = Expect.equal actual expected message
-    let inline notEqual actual expected message = Expect.notEqual actual expected message
```
Comment (HLWeil) [inline tests/TestingUtils/Library.fs:55]: Why? This caused errors for me previously.

Comment (HLWeil) [reply tests/TestingUtils/Library.fs:55]: In js

File: src/ISA/ISA/ArcTypes/CompositeHeader.fs [hunk 2]
```diff
@@ -97,6 +101,34 @@ type CompositeHeader =
...
+    /// 0: Expects no input
+    ///
+    /// 1: Expects OntologyAnnotation as input
+    ///
+    /// 2: Expects IOType as input
+    ///
+    /// 3: Expects string as input
+    /// </summary>
+    /// <param name="inp">Can be accessed from `CompositeHeader.Cases`</param>
+    static member jsGetColumnMetaType(inp:int) =
```
Comment (HLWeil) [inline src/ISA/ISA/ArcTypes/CompositeHeader.fs:120]: Sry, I don't quite get this. Could you further describe this in the issue?

**Review by HLWeil — APPROVED** [review]

File: tests/TestingUtils/Library.fs (hunk 1 above)
Comment (Freymaurer) [reply tests/TestingUtils/Library.fs:55]: inline decreases performance, and it is only required for equal.

File: src/ISA/ISA/ArcTypes/CompositeHeader.fs (hunk 2 above)
Comment (Freymaurer) [reply src/ISA/ISA/ArcTypes/CompositeHeader.fs:120]: programmatically decide how to create a specific CompositeHeader while looping over "Cases"

```js
it('jsGetColumnMetaType', function () {
        let cases = CompositeHeader.Cases
        let oa = OntologyAnnotation.fromString("My OA Name")
        let iotype = new IOType(0, [])
        let stringExample = "My Example"
        for (let mycase of cases) {
            let tag = mycase[0]
            let code = CompositeHeader.jsGetColumnMetaType(tag)
            switch (code) {
                case 0:
                    let header1 = new CompositeHeader(tag, [])
                    equal((header1.IsSingleColumn || header1.IsFeaturedColumn), true);
                    break;
                case 1:
                    let header2 = new CompositeHeader(tag, [oa])
                    equal(header2.IsTermColumn, true);
                    break;
                case 2:
                    let header3 = new CompositeHeader(tag, [iotype])
                    equal(header3.IsIOType, true);
                    break;
                case 3:
                    let header4 = new CompositeHeader(tag, [stringExample])
                    equal(header4.isFreeText, true);
                    break;
            }
        }
    })
```

**Pull request merged**